### PR TITLE
list of all goods in the hoods

### PR DESCRIPTION
**Pull request by Kavster92**
[DEMO LINK](https://Kavster92.github.io/react_list-of-goods/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -26,70 +29,121 @@ type ReorderOptions = {
...
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  if (sortType === SortType.ALPHABET) {
```
Comment (Grayfox147) [inline src/App.tsx:38]:
```suggestion
  if (sortType) {
    visibleGoods.sort((a, b) => {
      switch (sortType) {
        case SortType.ALPHABET:
          return a.localeCompare(b);

        case SortType.LENGTH:
          return a.length - b.length;

        default:
          return 0;
      }
    });
  }
```

**Review by Grayfox147 — COMMENTED** [review]
Overall a great work, really smart implementation! I left a couple of suggestions!

File: src/App.tsx (hunk 1 above)
Comment (naziamoff) [reply src/App.tsx:38]: sure, you always use switch if the number of conditions is limited

File: src/App.tsx [hunk 2]
```diff
@@ -26,70 +29,121 @@ type ReorderOptions = {
...
+              'button is-info',
+              {
+                'is-light': !isReversed,
+              },
+            )}
+            onClick={this.reverse}
+          >
+            Reverse
+          </Button>
+          {(isReversed || sortType !== SortType.NONE) && (
```
[the remark this answers is not in the document]
Comment (naziamoff) [reply src/App.tsx:135]: agree with @Grayfox147 on this one

**Review by naziamoff — CHANGES_REQUESTED** [review]

**Review by meforoma — APPROVED** [review]
✅

**Pull request closed without merging**